source/core.py

```diff
@@ -303,6 +303,9 @@ def main():
 	import speech
 	log.debug("Initializing speech")
 	speech.initialize()
+	from speech import sayAll
+	log.debug("Initializing sayAllHandler")
+	sayAll.SayAllHandler.initializeSpeechWithoutPauses()

```

Comment:
I think if we are going this far, then rather than make SayAllHandler a singleton, just construct it in core (passing speech function in). Other module that need it can be initialised after this and have the SayAllHandler instance passed to them, or if that is difficult with there current design, they can fetch it (at function level, not module level) from core.

Reply:
would it make more sense to do this in `speech.initialize()` so that they can fetch it from speech, not core?

Reply:
See 01b8c5f07